pkg/guardian/redis_conf_store.go

```diff
@@ -42,8 +47,17 @@ func NewRedisConfStore(redis *redis.Client, defaultWhitelist []net.IPNet, defaul
 		limit:           defaultLimit,
 		reportOnly:      defaultReportOnly,
 		routeRateLimits: make(map[url.URL]Limit),
-	}
-	return &RedisConfStore{redis: redis, logger: logger, conf: &lockingConf{conf: defaultConf}, reporter: mr,}
+		jails:           make(map[url.URL]Jail),
```

Comment:
Do we know the rough upper capacity limit for Go map lookups to maintain our latency requirements? It's probably much higher than we'll ever need to be concerned about, but it would be good to know the ballpark where other memory structures might need to be considered.

Reply:
The go map implementation does introduce a random seed to help avoid hash collisions. 

Go maps also keep track of the load factor to determine when the map should double in size, to also help avoid hash collisions. 

So, while I don't have any metrics to back this up at the moment, it does seem these map lookups should be quicker than the lookups for the whitelist or blacklist already being done by Guardian. 

I think it would be helpful for us to have some automated benchmarks for Guardian to help quantify the impact of changes like this.

Helpful reference:
https://dave.cheney.net/2018/05/29/how-the-go-runtime-implements-maps-efficiently-without-generics

Reply:
https://stackoverflow.com/a/29690063

It may be worth implementing the prisoner map with a `[4]byte` key (IP in byte array form) instead of a string, which would be a finite domain and thus lookups of O(1) versus a string (infinite domain) which would then probably average O(N log N).

This also lets us avoid doing string conversion of the IP in the hot path.

Again, these are probably going to be insignificant unless we hit millions of prisoners but worth thinking about now.

Reply:
> It may be worth implementing the prisoner map with a [4]byte key (IP in byte array form) instead of a string, which would be a finite domain 

Great idea, will update this PR.